server/events/vcs/gitea/client.go

```diff
@@ -0,0 +1,468 @@
+		Style:                  gitea.MergeStyleMerge,
+		Title:                  "Atlantis merge",
+		Message:                "Automatic merge by Atlantis",
+		DeleteBranchAfterMerge: pullOptions.DeleteSourceBranchOnMerge,
+		ForceMerge:             false,
+		HeadCommitId:           pull.HeadCommit,
+		MergeWhenChecksSucceed: false,
+	}
+
+	succeeded, resp, err := c.giteaClient.MergePullRequest(pull.BaseRepo.Owner, pull.BaseRepo.Name, int64(pull.Num), mergeOptions)
+
+	if err != nil {
+		return err
+	}
+
+	if !succeeded {
+		return fmt.Errorf("merge failed: %s", resp.Status)
+	}
+
+	return nil
+}
+
+// MarkdownPullLink specifies the string used in a pull request comment to reference another pull request.
+func (c *GiteaClient) MarkdownPullLink(pull models.PullRequest) (string, error) {
+	return fmt.Sprintf("#%d", pull.Num), nil
+}
+
+// GetTeamNamesForUser returns the names of the teams or groups that the user belongs to (in the organization the repository belongs to).
+func (c *GiteaClient) GetTeamNamesForUser(repo models.Repo, user models.User) ([]string, error) {
+	// TODO: implement
+	return nil, errors.New("GetTeamNamesForUser not (yet) implemented for Gitea client")
+}
+
+// GetFileContent a repository file content from VCS (which support fetch a single file from repository)
+// The first return value indicates whether the repo contains a file or not
+// if BaseRepo had a file, its content will placed on the second return value
+func (c *GiteaClient) GetFileContent(pull models.PullRequest, fileName string) (bool, []byte, error) {
+	content, _, err := c.giteaClient.GetContents(pull.BaseRepo.Owner, pull.BaseRepo.Name, pull.HeadCommit, fileName)
+
+	if err != nil {
+		return false, nil, err
+	}
+
+	if content.Type == "file" {
+		decodedData, err := base64.StdEncoding.DecodeString(*content.Content)
+		if err != nil {
+			return true, []byte{}, err
+		}
+		return true, decodedData, nil
+	}
+
+	return false, nil, nil
+}
+
+// SupportsSingleFileDownload returns true if the VCS supports downloading a single file
+func (c *GiteaClient) SupportsSingleFileDownload(repo models.Repo) bool {
+	return true
+}
+
+// GetCloneURL returns the clone URL of the repo
+func (c *GiteaClient) GetCloneURL(_ models.VCSHostType, repo string) (string, error) {
+	parts := strings.Split(repo, "/")
+	if len(parts) < 2 {
+		return "", errors.New("invalid repo format, expected 'owner/repo'")
+	}
+	repository, _, err := c.giteaClient.GetRepo(parts[0], parts[1])
+	if err != nil {
+		c.logger.Debug("GET /repos/%v/%v returned an error: %v", parts[0], parts[1], err)
+		return "", err
+	}
+	return repository.CloneURL, nil
+}
+
+// GetPullLabels returns the labels of a pull request
+func (c *GiteaClient) GetPullLabels(repo models.Repo, pull models.PullRequest) ([]string, error) {
+	page := 0
+	nextPage := 1
+	results := make([]string, 0)
+
+	opts := gitea.ListLabelsOptions{
+		ListOptions: gitea.ListOptions{
+			Page:     0,
+			PageSize: 100,
```

Comment:
hard coded, please see previous comments

Reply:
Added --gitea-page-size, defaulting to 30 based on Gitea docs.